### PR TITLE
rsa_kmgmt: Return OSSL_PKEY_PARAM_DEFAULT_DIGEST for unrestricted PSS keys

### LINKED ISSUE #13957: RSA-PSS does not have a default digest in master
In 1.1.1, performing a digest-sign operation without specifying a digest will cause OpenSSL to attempt to use the default digest for the specified algorithm. This works fine for RSA-PSS in that branch.

However, in master the logic does not seem to be there - and that seems deliberate for some reason.

The logic for 1.1.1 for determining the default digest is here:

https://github.com/openssl/openssl/blob/a7222fc14d5977210d2b4673a68039824a039dc2/crypto/rsa/rsa_ameth.c#L511-L523

In master this is where we might expect it to be handled:

https://github.com/openssl/openssl/blob/b897b353dff8138aa838bae9766ecd3de8c03280/providers/implementations/keymgmt/rsa_kmgmt.c#L314-L322

As you can see there is deliberate logic and a comment that provides a default digest for normal RSA, but for RSA-PSS it is skipped.

This is the cause of the failure noted here in https://github.com/openssl/openssl/issues/13931#issuecomment-766897892

### PR DESCRIPTION
**Pull request by t8m**
Fixes #13957

### CODE REVIEW

**Review by levitte — APPROVED** [review]

File: providers/implementations/keymgmt/rsa_kmgmt.c [hunk 1]
```diff
@@ -312,18 +312,20 @@ static int rsa_get_params(void *key, OSSL_PARAM params[])
...
-     * For RSA-PSS keys, we ignore the default digest request
-     * TODO(3.0) with RSA-OAEP keys, this may need to be amended
+     * For restricted RSA-PSS keys, we ignore the default digest request.
+     * With RSA-OAEP keys, this may need to be amended.
      */
     if ((p = OSSL_PARAM_locate(params, OSSL_PKEY_PARAM_DEFAULT_DIGEST)) != NULL
-        && rsa_type != RSA_FLAG_TYPE_RSASSAPSS) {
+        && (rsa_type != RSA_FLAG_TYPE_RSASSAPSS
+            || (rsa_type == RSA_FLAG_TYPE_RSASSAPSS
+                && ossl_rsa_pss_params_30_is_unrestricted(pss_params)))) {
```
Comment (levitte) [inline providers/implementations/keymgmt/rsa_kmgmt.c:321]: Side note: this is an equivalent expression:

``` C
        && (rsa_type != RSA_FLAG_TYPE_RSASSAPSS
            || ossl_rsa_pss_params_30_is_unrestricted(pss_params))) {
```

Comment (t8m) [reply providers/implementations/keymgmt/rsa_kmgmt.c:321]: true, will amend. I want to also add a test case.

Comment (levitte) [reply providers/implementations/keymgmt/rsa_kmgmt.c:321]: Ah, perhaps my approval was a bit premature then.  I'll dismiss it

**Comment by t8m** [comment]
@levitte please re-review

**Review by levitte — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master. Thank you for the review!

**Pull request closed without merging**